tests/test_animation.cpp

```diff
@@ -9,27 +9,30 @@ BOOST_AUTO_TEST_SUITE(AnimationTests)
 #if RW_TEST_WITH_DATA
 BOOST_AUTO_TEST_CASE(test_matrix) {
     {
-        Animation animation;
+        auto animation = std::make_shared<Animation>();
 
         /** Models are currently needed to relate animation bones <=> model
          * frame #s. */
         auto test_model = Global::get().d->loadClump("player.dff");
 
         Animator animator(test_model);
 
-        animation.duration = 1.f;
-        animation.bones["player"] = new AnimationBone{
-            "player",
-            0,
-            0,
-            1.0f,
-            AnimationBone::RT0,
-            {
-                {glm::quat(), glm::vec3(0.f, 0.f, 0.f), glm::vec3(), 0.f, 0},
-                {glm::quat(), glm::vec3(0.f, 1.f, 0.f), glm::vec3(), 1.0f, 1},
-            }};
-
-        animator.playAnimation(0, &animation, 1.f, false);
+        animation->duration = 1.f;
+        animation->bones["player"] = std::shared_ptr<AnimationBone>(
```

Comment:
make_shared?

Reply:
Does this need to be shared? Is there a risk of bones being used out of scope if they are owned by the animation?

Reply:
I made it a shared_pointer so it would be deleted automatically when the map would be destroyed (and to avoid adding a destructor).

I restored the plain pointer and added a destructor in the extra commit.

Reply:
I think he means could it not be unique_ptr - i.e its not actually shared ownership?

Reply:
The main reason I went for `std::shared_pointer<AnimationBone>` is because the [AnimationBone](https://github.com/rwengine/openrw/blob/11e90c61e56b60240251cd080f175ddff7d7a101/rwengine/src/loaders/LoaderIFP.hpp#L41) is referenced as key in [AnimationState](https://github.com/rwengine/openrw/blob/11e90c61e56b60240251cd080f175ddff7d7a101/rwengine/src/engine/Animator.hpp#L37).